zzk/connection.go

```diff
@@ -38,6 +40,34 @@ var (
 // GetConnection describes a generic function for acquiring a connection object
 type GetConnection func(string) (client.Connection, error)
 
+type delay struct {
+	backoff time.Duration
+}
+
+func (d *delay) GetDelay() time.Duration {
+	defer func() {
+		factor := 1.25
+		jitter := 0.1
+
+		d.backoff = time.Duration(float64(d.backoff) * factor)
+		if d.backoff > MaxDelay {
+			d.backoff = MaxDelay
+		}
+		d.backoff += time.Duration(rand.NormFloat64() * jitter * float64(time.Second))
```

Comment:
This value will stop when it is greater than MaxBackoff
We want it to always be less than or equal to MaxDelay